astropy/coordinates/sky_coordinate.py

```diff
@@ -1486,22 +1486,50 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
         care is needed in application. Strictly speaking, the barycentric correction is
         multiplicative and should be applied as::
 
-           sc = SkyCoord(1*u.deg, 2*u.deg)
-           vcorr = sc.rv_correction(kind='barycentric', obstime=t, location=loc)
-           rv = rv + vcorr + rv * vcorr / consts.c
+          >>> from astropy.time import Time
+          >>> from astropy.coordinates import SkyCoord, EarthLocation
+          >>> t = Time(56370.5, format='mjd', scale='utc')
+          >>> loc = EarthLocation('149d33m00.5s','-30d18m46.385s',236.87*u.m)
+          >>> sc = SkyCoord(1*u.deg, 2*u.deg)
+          >>> vcorr = sc.radial_velocity_correction(kind='barycentric', obstime=t, location=loc)
+          >>> rv = rv + vcorr + rv * vcorr / consts.c
 
         If your target is nearby and/or has finite proper motion you may need to account
-        for terms arising from this. See Wright & Eastmann (2014) for details.
+        for terms arising from this. See Wright & Eastman (2014) for details.
+
+        Also note that this method returns the correction velocity in the so-called
+        *optical convention*::
+
+          >>> vcorr = zb * consts.c
+
+        where ``zb`` is the barycentric correction redshift as defined in section 3
+        of Wright & Eastman (2014). The application formula given above follows from their
+        equation (11) under assumption that the radial velocity ``rv`` has also been defined
+        using the same optical convention. Note, this can be regarded as a matter of
+        velocity definition and does not by itself imply any loss of accuracy, provided
+        sufficient care has been taken during interpretation of the results. If you need
+        the barycentric correction expressed as the full relativistic velocity (e.g., to provide
+        it as the input to another software which performs the application), the
+        following recipe can be used::
+
+          >>> zb = vcorr / consts.c
+          >>> zb_plus_one_squared = (zb + 1) ** 2
+          >>> vcorr_rel = consts.c * (zb_plus_one_squared - 1) / (zb_plus_one_squared + 1)
```

Comment:
There is a way to do this using *just* equivalencies:

```
>>> vcorr_rel = vcorr.to(u.Hz, u.doppler_optical(1*u.Hz)).to(vcorr.unit, u.doppler_relativistic(1*u.Hz))
```
I'd say best to switch to that *except* that it's sort of un-intuitive that it requires a choice of "rest" frequency (which doesn't matter in this case as long as you use the same one for both equivalencies).

Perhaps it's best to use what I have written there because it's more concise and units-complient, but then add "or more expressively:" and then what you have here?

Reply:
thanks, @eteq. Your suggestions make sense (although for the second code snippet, I'd perhaps give the explicit formula first as this is what I've tested extensively as part of my science work and what helps relating the result to the notations used in Wright & Eastman's paper cited earlier. Your way of doing it via equivalencies can be given later as an alternative). 

I've added the imports at some later iteration when CI started to run these examples (they were just text before) and just did the minimum addition which worked (perhaps because consts were imported through some other import?). I don't have any problem with importing c explicitly. 

Do I need to rebase again before making these changes (given that it took a lot of time since the previous commit)?